pandas/core/arrays/_mixins.py

```diff
@@ -304,6 +304,48 @@ def __repr__(self) -> str:
     # ------------------------------------------------------------------------
     # __array_function__ methods
 
+    def __array_function__(self, func, types, args, kwargs):
+
+        if not args:
+            # I dont think this is possible is it?
+            raise NotImplementedError
```

Comment:
This doesn't seem right. Types cannot be empty, args probably can be if kwargs is used only.

You should just have an early check for whether you have to always defer, using something like:

```
for t in types:
    if t not in {np.ndarray, ?} or issubclass(t, NDArrayBackedExtensionArray):
        return NotImplemented
```
or something similar.  For things like `delete` types will always just be `(NDArrayBackedExtensionArray,)` (or the subclass).

Reply:
To be clear, I think this will break (not that it is typical syntax):
```
np.delete(arr=np.arange(5), obj=4)
```
But checking all types is necessary, since you only want to convert `NDArrayBackedExtensionArray` to a numpy array in the fallback, if there is a Dask array involved, Dask will have to deal with `NDArrayBackedExtensionArray` probably.

Reply:
> To be clear, I think this will break (not that it is typical syntax):

yes it does, good catch

Reply:
do we need to worry about self being nested among kwargs? (like we do for it being nested in one of args)

Reply:
> do we need to worry about self being nested among kwargs? (like we do for it being nested in one of args)

i guess we do, same way of cooking up an example as you did for np.delete